Yoast/wordpress-seo
# Adapt meta description keyword research for Japanese

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

[yoastseo] Adapts metaDescriptionKeyword research for Japanese and moves the check for Japanese feature flag to the main function in getMorphologyData.js helper.

## Relevant technical choices:

* The check for Japanese feature flag in `getMorphologyData.js` helper was initially done outside of the `getMorphologyData` function. If the flag is enabled, the Japanese morphology data should be added to the morphology data array. However, it turns out that the feature flag is never going to be recognized as enabled outside of the function in `getMorphologyData.js`, even though the flag is enabled when running the unit tests for example. As the result, Japanese data was never added to the array.
* The problem above is solved by moving the check for the Japanese feature flag is moved to inside of the main function.
* The sentence `Returns -1 if no description is specified in the given paper.` is removed for the `metaDescriptionKeyword `function description because the function doesn't return -1 if there is no description being set in the paper, hence irrelevant.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

**In content-analysis app (to test different word forms)**
* Enable `JAPANESE_SUPPORT` in `specHelpers/getMorphologyData.js` and in `packages/yoastseo/src/languageProcessing/helpers/sentence/getSentences.js` (i.e. `enableFeatures( [ "JAPANESE_SUPPORT" ] ); `(don't commit anything)
* Start the app (run `yarn start` in wordpress-seo/apps/content-analysis)
* Set the locale to` ja`
* Turn the Use morphology toggle on
* Create a post with at least 15 sentences (use[ this generator](https://generator.lorem-ipsum.info/_japanese) to generate Japanese text)
* Set `小さい花の刺繍` as the keyphrase
* Add this sentence `小さくて可愛い花の刺繍に関する一般一般の記事です。` to the meta description field
* Confirm that the Keyphrase in meta description assessment returns with green bullet with the following feedback:

`Keyphrase or synonym appear in the meta description. Well done!`

**In Docker (to test the feature flag)**

* Build Free plugin
* Run` composer require yoast/wordpress-seo:dev-LINGO-1103-adapt-metaDescriptionKeyword-research-for-Japanese@dev` in wordpress-seo-premium.
* Build the premium plugin
* Set the site language to Japanese
* Test with `JAPANESE_SUPPORT enabled`: Enable the` JAPANESE_SUPPORT` feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', true );` in `basic-wordpress-config.php `file
* Create a post
* Set the keyphrase to `かわいらしいべっ甲猫`
* Add this sentences `この記事は小さい花の刺繍をどうやってすてればいいのか、基本的な情報を紹介します。私は美しい猫を飼っています。 `to the meta description field
* Confirm that the Keyphrase in meta description assessment returns with red bullet with the following feedback:
````
メタディスクリプション中のキーフレーズ: メタディスクリプションが設定されていますが、キーフレーズがありません。修正しましょう !
````
* Set `小さい花の刺繍` as the keyphrase
* Confirm that the Keyphrase in meta description assessment returns with green bullet and following feedback:
````
メタディスクリプション中のキーフレーズ: メタディスクリプション中にキーフレーズまたは同義語が含まれています。いいですね !
````
* Change the keyphrase to `野生のハーブの刺繡`
* Add this phrase as the synonyms `小さい花の刺繍`
Confirm that the Keyphrase in meta description assessment still returns with green bullet and give the following feedback:
````
メタディスクリプション中のキーフレーズ: メタディスクリプション中にキーフレーズまたは同義語が含まれています。いいですね !
````
* Add this sentences `野生のハーブの刺繡のためのいくつかのヒント。野生のハーブの刺繡のためのいくつかのヒント。 `to the meta description field
* Confirm that the Keyphrase in meta description assessment returns with red bullet with the following feedback:
````
メタディスクリプション中のキーフレーズ: メタディスクリプション中にキーフレーズが3回含まれており、おすすめ上限の2回を超過しています。制限しましょう !
````

* **Test with JAPANESE_SUPPORT disabled**

* Disable the JAPANESE_SUPPORT feature by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', false );` in` basic-wordpress-config.php` file
* Save your changes in the previous post and update it
* Confirm that the Keyphrase in meta description assessment returns with red bullet with the following feedback:
````
メタディスクリプション中のキーフレーズ: メタディスクリプションが設定されていますが、キーフレーズがありません。修正しましょう !
````


### Test instructions for QA when the code is in the RC


 Test the `In Docker (to test the feature flag) ` part
 
## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-1103
